packages/react-native-quick-crypto/src/cipher.ts

```diff
@@ -0,0 +1,292 @@
+    const res = this.native.setAuthTag(binaryLikeToArrayBuffer(tag));
+    if (!res) {
+      throw new Error('setAuthTag failed');
+    }
+    return this;
+  }
+
+  public getSupportedCiphers(): string[] {
+    return this.native.getSupportedCiphers();
+  }
+}
+
+class Cipheriv extends CipherCommon {
+  constructor(
+    cipherType: string,
+    cipherKey: BinaryLikeNode,
+    iv: BinaryLike,
+    options: Record<string, TransformOptions> = {},
+  ) {
+    super({
+      isCipher: true,
+      cipherType,
+      cipherKey: binaryLikeToArrayBuffer(cipherKey),
+      iv: binaryLikeToArrayBuffer(iv),
+      options,
+    });
+  }
+}
+
+type Cipher = Cipheriv;
+
+class Decipheriv extends CipherCommon {
+  constructor(
+    cipherType: string,
+    cipherKey: BinaryLikeNode,
+    iv: BinaryLike,
+    options: Record<string, TransformOptions> = {},
+  ) {
+    super({
+      isCipher: false,
+      cipherType,
+      cipherKey: binaryLikeToArrayBuffer(cipherKey),
+      iv: binaryLikeToArrayBuffer(iv),
+      options,
+    });
+  }
+}
+
+type Decipher = Decipheriv;
+
+export function createDecipheriv(
+  algorithm: CipherCCMTypes,
+  key: BinaryLikeNode,
+  iv: BinaryLike,
+  options: CipherCCMOptions,
+): Decipher;
+export function createDecipheriv(
+  algorithm: CipherOCBTypes,
+  key: BinaryLikeNode,
+  iv: BinaryLike,
+  options: CipherOCBOptions,
+): Decipher;
+export function createDecipheriv(
+  algorithm: CipherGCMTypes,
+  key: BinaryLikeNode,
+  iv: BinaryLike,
+  options?: CipherGCMOptions,
+): Decipher;
+export function createDecipheriv(
+  algorithm: CipherType,
+  key: BinaryLikeNode,
+  iv: BinaryLike,
+  options?: Stream.TransformOptions,
+): Decipher;
+export function createDecipheriv(
+  algorithm: string,
+  key: BinaryLikeNode,
+  iv: BinaryLike,
+  options?:
+    | CipherCCMOptions
+    | CipherOCBOptions
+    | CipherGCMOptions
+    | Stream.TransformOptions,
+): Decipher {
+  return new Decipheriv(
+    algorithm,
+    key,
+    iv,
+    options as Record<string, TransformOptions>,
+  );
+}
+
+export function createCipheriv(
+  algorithm: CipherCCMTypes,
+  key: BinaryLikeNode,
+  iv: BinaryLike,
+  options: CipherCCMOptions,
+): Cipher;
+export function createCipheriv(
+  algorithm: CipherOCBTypes,
+  key: BinaryLikeNode,
+  iv: BinaryLike,
+  options: CipherOCBOptions,
+): Cipher;
+export function createCipheriv(
+  algorithm: CipherGCMTypes,
+  key: BinaryLikeNode,
+  iv: BinaryLike,
+  options?: CipherGCMOptions,
+): Cipher;
+export function createCipheriv(
+  algorithm: CipherType,
+  key: BinaryLikeNode,
+  iv: BinaryLike,
+  options?: Stream.TransformOptions,
+): Cipher;
+export function createCipheriv(
+  algorithm: string,
+  key: BinaryLikeNode,
+  iv: BinaryLike,
+  options?:
+    | CipherCCMOptions
+    | CipherOCBOptions
+    | CipherGCMOptions
+    | Stream.TransformOptions,
+): Cipher {
+  return new Cipheriv(
+    algorithm,
+    key,
+    iv,
+    options as Record<string, TransformOptions>,
```

Comment:
Is this casting safe?

I see the argument is of type `TransformOptions`, and this is casting it to an object of `TransformOptions`